rust/arrow/src/buffer.rs

```diff
@@ -94,13 +97,40 @@ impl Debug for BufferData {
 }
 
 impl Buffer {
-    /// Creates a buffer from an existing memory region (must already be byte-aligned)
+    /// Creates a buffer from an existing memory region (must already be byte-aligned), and this
+    /// buffer will free this piece of memory when dropped.
+    ///
+    /// This method is the same as `from_owned`.
     pub fn from_raw_parts(ptr: *const u8, len: usize) -> Self {
+        Buffer::build_with_arguments(ptr, len, true)
+    }
+
+    /// Creates a buffer from an existing memory region (must already be byte-aligned), and this
+    /// buffer will free this piece of memory when dropped.
+    pub fn from_owned(ptr: *const u8, len: usize) -> Self {
```

Comment:
Should we just keep one method? Having two method with the same name is slightly confusing. For instance, someone may use both methods in their code for some reason which makes reading more difficult. 

I think the naming is not an issue as long as we make sure the method is well-documented.

Reply:
I agree that two methods with same functionality is slightly confusing. But I think `from_owned` is better than `from_raw_parts` because this naming style is more consistent with `from_unowned`, that's also why I deprecate `from_raw_parts` at first. If we should only keep one, I'd prefer to keep `from_owned`.

Reply:
Yes,`from_owned` sounds slightly better than `from_raw_parts`, but I don't think it's worth having two methods with the same functionality (which will cause confusion), or deprecating the current one. In fact taking ownership of the incoming memory should be the default behavior so this `owned` part is also not really necessary.

Reply:
Fixed.